### PR TITLE
8272377: assert preconditions that are ensured when created in add_final_edges

### PR DESCRIPTION
**Pull request by kelthuzadx**
I think we can assert preconditions, which are ensured by add_node_to_connection_graph when they were first added to CG, in add_final_edges when they are delay handled, then we can skip checking these preconditions in runtime(assertions instead). This PR also does some code refactor.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272377](https://bugs.openjdk.java.net/browse/JDK-8272377): assert preconditions that are ensured when created in add_final_edges

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5101/head:pull/5101` \
`$ git checkout pull/5101`

Update a local copy of the PR: \
`$ git checkout pull/5101` \
`$ git pull https://git.openjdk.java.net/jdk pull/5101/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5101`

View PR using the GUI difftool: \
`$ git pr show -t 5101`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5101.diff">https://git.openjdk.java.net/jdk/pull/5101.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/escape.cpp
Comment (TobiHartmann) [inline src/hotspot/share/opto/escape.cpp:877]: Indentation is wrong.

File: src/hotspot/share/opto/escape.cpp
Comment (TobiHartmann) [inline src/hotspot/share/opto/escape.cpp:166]: Missing whitespace after `switch`.

File: src/hotspot/share/opto/escape.cpp
Comment (TobiHartmann) [inline src/hotspot/share/opto/escape.cpp:698]: I would change the assert message to `Unexpected node type`. Same for other asserts below.

**Review by TobiHartmann — CHANGES_REQUESTED** [review]
Nice refactoring. I verified that the conditions are indeed checked in `ConnectionGraph::add_node_to_connection_graph`.

I added some minor comments, looks good to me otherwise.

File: src/hotspot/share/opto/escape.cpp
Comment (kelthuzadx) [reply src/hotspot/share/opto/escape.cpp]: Thanks for the review, Tobias! All changed.

**Review by TobiHartmann — APPROVED** [review]
Looks good. You need a second review.

**Review by vnkozlov — APPROVED** [review]
I am fine with refactoring: using `switch()` and passing `mergemem_worklist` as argument.

Using `ELSE_FAIL()` macro was very helpful during development and I kept it.
But we did not see such failure for long time so I am fine with replacing it with just assert.

Testing ran by Tobias was clean.

**Comment by kelthuzadx** [comment]
Thanks for the review, Vladimir K. !

**Pull request closed without merging**
